src/nothrow.jl

```diff
@@ -248,28 +248,38 @@ will fail (i.e., return a `NoThrowResult{Missing}` if:
 In any of these failure cases, this function will not throw, but instead will return
 the cause of failure in the output `violations` field.
 
+If `verbose_violations=true`, then much more verbose violation strings will be generated in the case of unexpected violations (including full stacktraces).
+
 !!! note
   For debugging purposes, it may be helpful to bypass the "no-throw" feature and
   so as to have access to a callstack. To do this, use [`transform_force_throw!`](@ref)
   in place of `transform!`.
 
 See also: [`convert_spec`](@ref)
 """
-function transform!(ntt::NoThrowTransform, input)
+function transform!(ntt::NoThrowTransform, input; verbose_violations=false)
     # Check that input meets specification
     InSpec = input_specification(ntt)
     input = try
         convert_spec(InSpec, input)
     catch e
-        violations = "Input doesn't conform to specification `$(InSpec)`. Details: $e"
+        if verbose_violations
+            stack = current_exceptions()
+            str = sprint(show, MIME"text/plain"(), stack)
+        else
+            str = string(e)
+        end
+        violations = "Input doesn't conform to specification `$(InSpec)`. Details: $str"
         return NoThrowResult(; violations)
     end
 
     # Do transformation
     result = try
         NoThrowResult(ntt.transform_spec.transform_fn(input))
-    catch e
-        return NoThrowResult(; violations="Unexpected violation. Details: $e")
+    catch
+        stack = current_exceptions()
```

Comment:
I wonder if this also should be switched with `verbose_violations = true` and `string(e)` used otherwise.

Reply:
it is a `NoThrowResult` call which suggests there is no exception I guess.

Reply:
oops, I meant to check this one too

Reply:
good catch! pushed a fix